ttnn/cpp/ttnn/operations/moreh/moreh_arange/device/moreh_arange_device_operation.cpp

```diff
@@ -0,0 +1,117 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "moreh_arange_device_operation.hpp"
+
+#include <cstdint>
+
+#include "tt_dnn/op_library/moreh_helper_functions.hpp"
+#include "ttnn/tensor/tensor.hpp"
+
+namespace ttnn::operations::moreh::moreh_arange {
+void MorehArangeOperation::validate_inputs(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    auto step = operation_attributes.step;
+    auto start = operation_attributes.start;
+    auto end = operation_attributes.end;
+    TT_ASSERT(step != 0, "step must be nonzero.");
+    TT_ASSERT(
```

Comment:
TT_ASSERT only happens in debug builds. 
Please use TT_FATAL for user input validation for consistency.